src/framework/uicomponents/view/validators/intinputvalidator.cpp

```diff
@@ -48,8 +48,10 @@ QValidator::State IntInputValidator::validate(QString& inputStr, int& cursorPos)
 {
     QValidator::State state = Invalid;
 
-    if (inputStr.contains(QRegularExpression("^\\-?\\d{1,3}$"))) {
-        if (inputStr.contains(QRegularExpression("^\\-?0{2,3}"))
+    // The expression is based on the number of digits in the maximum absolute value between m_top and m_bottom.
+    if (inputStr.contains(QRegularExpression(QString("^\\-?\\d{1,%1}$").arg(std::ceil(std::log10(std::max(std::abs(m_top),
+                                                                                                          std::abs(m_bottom)))))))) {
+        if (inputStr.contains(QRegularExpression("^\\-?0{2,4}"))
```

Comment:
The '4' should be replaced here too. And it would be better to extract that intimidating std::ceil(………) expression into a named local variable.

Reply:
Yes! the if statement looks way too long, I'll make the necessary changes and commit soon.